great_expectations/data_context/util.py

```diff
@@ -205,7 +248,208 @@ def substitute_config_variable(
+
+    - Azure Key Vault: ``secret|https://vault-name.vault.azure.net/secrets/my-secret``
+
+    :param value: a string that might or might not start with `secret|`
+
+    :return: a string with the value substituted by the secret from the secret store,
+            or the same object if value is not a string.
+    """
+    if isinstance(value, str) and value.startswith("secret|"):
+        if value.startswith("secret|arn:aws:secretsmanager"):
+            return substitute_value_from_aws_secrets_manager(value)
+        elif GCP_SECRET_MANAGER_FAST_REGEX.match(value):
+            return substitute_value_from_gcp_secret_manager(value)
+        elif AZURE_KEYVAULT_FAST_REGEX.match(value):
+            return substitute_value_from_azure_keyvault(value)
+    return value
+
+
+def substitute_value_from_aws_secrets_manager(value):
+    """
+    This methods uses a boto3 client and the secretsmanager service to try to retrieve the secret value
+    from the elements it is able to parse from the input value.
+
+    - value: string with pattern ``secret|arn:aws:secretsmanager:${region_name}:${account_id}:secret:${secret_name}``
+
+        optional : after the value above, a secret version can be added ``:${secret_version}``
+
+        optional : after the value above, a secret key can be added ``|${secret_key}``
+
+    - region_name: `AWS region used by the secrets manager <https://docs.aws.amazon.com/general/latest/gr/rande.html>`_
+    - account_id: `Account ID for the AWS account used by the secrets manager <https://docs.aws.amazon.com/en_us/IAM/latest/UserGuide/console_account-alias.html>`_
+
+            This value is currently not used.
+    - secret_name: Name of the secret
+    - secret_version: UUID of the version of the secret
+    - secret_key: Only if the secret's data is a JSON string, which key of the dict should be retrieve
+
+    :param value: a string that starts with ``secret|arn:aws:secretsmanager``
+
+    :return: a string with the value substituted by the secret from the AWS Secrets Manager store
+
+    :raises: ImportError, ValueError
+    """
+    if not boto3:
+        logger.error(
+            "boto3 is not installed, please install great_expectations with aws_secrets extra > "
+            "pip install great_expectations[aws_secrets]"
+        )
+        raise ImportError("Could not import boto3")
+
+    matches = AWS_SECRET_MANAGER_SLOW_REGEX.match(value)
+
+    if not matches:
+        raise ValueError(
+            f"Could not match the value with regex {AWS_SECRET_MANAGER_SLOW_REGEX}"
+        )
+
+    region_name = matches.group(1)
+    secret_name = matches.group(3)
+    secret_version = matches.group(4)
+    secret_key = matches.group(5)
+
+    # Create a Secrets Manager client
+    session = boto3.session.Session()
+    client = session.client(service_name="secretsmanager", region_name=region_name)
+
+    secret_response = client.get_secret_value(
+        SecretId=secret_name, VersionId=secret_version
+    )
```

Comment:
```suggestion

    if secret_version:
        secret_response = client.get_secret_value(SecretId=secret_name, VersionId=secret_version)
    else:
        secret_response = client.get_secret_value(SecretId=secret_name)
```

-  I was getting an error in situations when `secret_name` is `None`